src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -98,10 +98,11 @@ protected override ModuleData CreateValueFromKey(string key)
 
         private SharedGenericsMode _genericsMode;
         
-        public CompilerTypeSystemContext(TargetDetails details, SharedGenericsMode genericsMode)
+        public CompilerTypeSystemContext(TargetDetails details, SharedGenericsMode genericsMode, bool supportsLazyCctors = false)
```

Comment:
ReadyToRun mode shouldn't call into places that deal with CoreRT's lazy static constructors. Lazy static constructors are a CoreRT concept and none of the associated behaviors (e.g. reporting `false` for the lazy static constructor check if the type has a custom attribute that makes the cctor eager) should light up on CoreCLR.

Looks like there's two places in CorInfoImpl - I would just ifdef those out and replace the implementation with what crossgen has there (probably just checking in the metadata and maybe dealing with the `<Module>` type?).

Reply:
Fixed in 5th iteration, sorry about having overlooked it in my previous change.